### PR TITLE
cleanup function to be more readable

### PR DESCRIPTION
**Pull request by stefanhayden**
This is not exactly fabric's style and a lot of this is personal preference but I find these changes make the function more readable and I believe would have made finding the bug we previously fixed a little easier.

### CODE REVIEW

File: src/mixins/text_style.mixin.js [hunk 1]
```diff
@@ -63,31 +63,42 @@
...
+          if (stylePropertyHasBeenSet) {
+            if (!style) {
+              style = styleObject[property];
+            }
+            else if (styleObject[property] !== style) {
+              allStyleObjectPropertiesMatch = false;
+            }
+
+            if (styleObject[property] === this[property]) {
+              delete obj[p1][p2][property];
```
Comment (asturur) [inline src/mixins/text_style.mixin.js]: this should be a reference to styleObject, should be fine to use it to delete too.

File: src/mixins/text_style.mixin.js [hunk 2]
```diff
@@ -63,31 +63,42 @@
       if (!this.styles || !property || property === '') {
         return false;
       }
-      var obj = this.styles, stylesCount = 0, letterCount, foundStyle = false, style,
-          canBeSwapped = true, graphemeCount = 0;
+      var obj = this.styles, stylesCount = 0, letterCount, style,
+          allStyleObjectPropertiesMatch = true, graphemeCount = 0, styleObject;
```
Comment (asturur) [inline src/mixins/text_style.mixin.js]: what do you think about:
style  => stylePropertyValue

Comment (stefanhayden) [reply src/mixins/text_style.mixin.js]: clearly I love verbose names!

File: src/mixins/text_style.mixin.js (hunk 1 above)
Comment (stefanhayden) [reply src/mixins/text_style.mixin.js]: true!

**Pull request merged**